src/metaxy/utils/hashing.py

```diff
@@ -0,0 +1,142 @@
+"""Hash truncation utilities for Metaxy.
+
+This module provides utilities for globally truncating hash outputs to reduce
+storage requirements and improve readability. Hash truncation is configured
+through the global MetaxyConfig.
+"""
+
+# Minimum allowed truncation length
+MIN_TRUNCATION_LENGTH = 8
+
+
+def truncate_hash(hash_str: str, length: int | None = None) -> str:
```

Comment:
this should not take length as an option, length be taken with `get_hash_truncation_length` 

also, a narwhals (`@narwhalify`) function doing the same with
- a column
- a struct (truncate values)

is needed